File: Flight-Reservation-System-Backend/src/main/java/com/flightreservation/service/BookingService.java
package com.flightreservation.service;

import java.util.List;

import com.flightreservation.dto.FlightDetailsDto;
import com.flightreservation.dto.FlightsDto;
import com.flightreservation.dto.PassengerDto;
import com.flightreservation.dto.SeatDto;
import com.flightreservation.model.FlightDetails;
import com.flightreservation.model.Seat;

public interface BookingService {

	List<FlightsDto> getAllFlightsAndSeatsDetails();

	List<FlightDetailsDto> getFlightByArrivalAndDeparture(String departure, String arrival, String departureDate);

	List<SeatDto> getAllSeatsFromAFlight(String flightNumber);

	PassengerDto bookFlightForAPassenger(FlightDetails flightDetails, int userId);
	
	String bookSeatForAPassenger(Seat seat, String flightNumber, int userId);

}

File: Flight-Reservation-System-Frontend/src/services/BookingService.js
import axios from "axios";

const BOOKING_API_BASE_URL = "http://localhost:8080/booking";

class BookingService {
  getFlightByArrivalAndDeparture(departure, arrival, departureDate) {
    return axios.get(
      BOOKING_API_BASE_URL +
        "/getFlightsOnArrivalAndDeparture/" +
        departure +
        "/" +
        arrival +
        "/" +
        departureDate
    );
  }

  getAllSeatsFromAFlight(flightNumber) {
    return axios.get(BOOKING_API_BASE_URL + "/getAllSeats/" + flightNumber);
  }

  bookFlightForAPassenger(userId, flightDetails) {
    return axios.post(
      BOOKING_API_BASE_URL + "/bookFlights/" + userId,
      flightDetails
    );
  }

  bookSeatForAPassenger(flightNumber, userId, seat) {
    return axios.post(
      BOOKING_API_BASE_URL + "/bookSeats/" + flightNumber + "/" + userId,
      seat
    );
  }
}

export default new BookingService();
